src/sfeQwiicUltrasonic.cpp

```diff
@@ -48,14 +79,21 @@ sfeTkError_t sfeQwiicUltrasonic::triggerAndRead(uint16_t &distance)
     return kSTkErrOk;
 }
 
-sfeTkError_t sfeQwiicUltrasonic::changeAddress(const uint8_t &address)
+sfeTkError_t sfeQwiicUltrasonic::changeAddress(uint8_t &address)
 {
     // Check whether the address is valid
+    sfeTkError_t err;
+    size_t numBytes = 2;
+    // We want to shift the address left before we send it. 
+    address <<= 1;
+    const uint8_t toWrite[2] = {kUltrasonicAddressChangeCommand, address};
+
     if (address < kQwiicUltrasonicMinAddress || address > kQwiicUltrasonicMaxAddress)
         return kSTkErrFail;
 
     // Write the new address to the device. The first bit must be set to 1
-    sfeTkError_t err = _theBus->writeByte(address | 0x80);
+    // err = _theBus->writeRegisterByte(kUltrasonicAddressChangeCommand, (address<< 1));
+    err = _theBus->writeBlock(toWrite, numBytes);
```

Comment:
This looks like a change in functionality, will this work for boards with the old firmware?

Reply:
Yeah that's a good point, it will not. I can add an additional function for backwards compatibility.

Reply:
I left a comment a little lower but I restored the `changeAddress()` function to its' previous iteration and then changed the name of the one I added to `updateAddress()`. I'll make sure to make these distinctions in the hookup guide and the examples.